ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/KafkaConfigStore.java

```diff
@@ -0,0 +1,188 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.server.computation;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import io.confluent.kafka.serializers.KafkaJsonDeserializer;
+import io.confluent.kafka.serializers.KafkaJsonSerializer;
+import io.confluent.ksql.rest.util.KsqlInternalTopicUtils;
+import io.confluent.ksql.util.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class KafkaConfigStore implements ConfigStore {
+  private static final Logger log = LoggerFactory.getLogger(KafkaConfigStore.class);
+
+  private static final String CONFIG_TOPIC_SUFFIX = "configs";
+  private static final String CONFIG_MSG_KEY = "ksql-standalone-configs";
+
+  private final KsqlConfig ksqlConfig;
+
+  static Deserializer<KsqlProperties> createDeserializer() {
+    KafkaJsonDeserializer<KsqlProperties> deserializer = new KafkaJsonDeserializer<>();
+    deserializer.configure(
+        ImmutableMap.of(
+            "json.fail.unknown.properties", false,
+            "json.value.type", KsqlProperties.class
+        ),
+        false
+    );
+    return deserializer;
+  }
+
+  private static KafkaConsumer<String, KsqlProperties> createConsumer(KsqlConfig ksqlConfig) {
+    return new KafkaConsumer<>(
+        ksqlConfig.getKsqlStreamConfigProps(),
+        Serdes.String().deserializer(),
+        createDeserializer());
+  }
+
+  private static KafkaProducer<String, KsqlProperties> createProducer(KsqlConfig ksqlConfig) {
+    final Serializer<KsqlProperties> serializer = new KafkaJsonSerializer<>();
+    serializer.configure(Collections.emptyMap(), false);
+    return new KafkaProducer<>(
+        ksqlConfig.getKsqlStreamConfigProps(),
+        Serdes.String().serializer(),
+        serializer);
+  }
+
+  public KafkaConfigStore(final KsqlConfig currentConfig, final KafkaTopicClient topicClient) {
+    try (
+        KafkaConsumer<String, KsqlProperties> consumer = createConsumer(currentConfig);
```

Comment:
nit: `final`

Reply:
`final` would be superfluous here (and I think we cause a check style failure).